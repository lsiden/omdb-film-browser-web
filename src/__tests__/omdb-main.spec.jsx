import React from 'react'
import TestRenderer from 'react-test-renderer'
import { StaticRouter } from 'react-router-dom'

// import { OmdbMainUnconnected as OmdbMain } from 'omdb-main'
import OmdbMain from 'omdb-main'

const createWrapper = () => TestRenderer.create(
  <StaticRouter context={{}}>
    <OmdbMain />
  </StaticRouter>,
)

describe('OmdbMain', () => {
  test('matches snapshot', () => {
    expect(createWrapper()).toMatchSnapshot()
  })
})
