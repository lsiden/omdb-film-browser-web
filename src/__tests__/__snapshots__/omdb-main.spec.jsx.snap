// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OmdbMain matches snapshot 1`] = `
<div>
  <div>
    <header
      style={
        Object {
          "background": "linear-gradient(to right, #000040, #3b00ff)",
          "color": "white",
          "padding": "10px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <h1>
          <img
            alt="OMDB Browser logo"
            height={20}
            src="movie-icon.svg"
            style={
              Object {
                "margin": 12,
              }
            }
          />
          Open Movie Database Browser
        </h1>
        <a
          href="/about"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          About
        </a>
      </div>
    </header>
    <form
      style={
        Object {
          "background": "linear-gradient(to right, #000040, #3b00ff)",
          "color": "white",
          "padding": "10px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <label
          htmlFor="query-form-search-input"
          style={
            Object {
              "marginRight": "1em",
            }
          }
        >
          Search
        </label>
        <div
          style={
            Object {
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <span
            charSet="utf-8"
            dangerouslySetInnerHTML={
              Object {
                "__html": "⚲",
              }
            }
            style={
              Object {
                "color": "#888",
                "fontSize": "20pt",
                "left": "6pt",
                "position": "absolute",
                "top": "-4pt",
                "transform": "rotate(-45deg)",
              }
            }
          />
          <input
            onChange={[Function]}
            placeholder="Film Title ..."
            style={
              Object {
                "color": "black",
                "fontSize": "12pt",
                "height": "24pt",
                "textIndent": "24pt",
                "width": "100%",
              }
            }
            type="search"
            value=""
          />
          <span
            aria-label="Cancel"
            charSet="utf-8"
            className="hover-pointer"
            dangerouslySetInnerHTML={
              Object {
                "__html": "⨉",
              }
            }
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            style={
              Object {
                "bottom": 4,
                "color": "#888",
                "fontSize": 16,
                "position": "absolute",
                "right": 8,
              }
            }
            tabIndex={0}
          />
        </div>
      </div>
    </form>
    <div
      style={
        Object {
          "color": "#210d87",
          "fontSize": "14pt",
          "margin": "1em",
        }
      }
    >
       Search for a title. 
    </div>
  </div>
</div>
`;
